cards/functions/retrieve-cards/main.go

```diff
@@ -0,0 +1,30 @@
+package main
+
+import (
+	"context"
+	clientModel "kargo-back/models/clients"
+	"kargo-back/shared/apigateway"
+	"kargo-back/shared/random"
+	storage "kargo-back/storage/cards"
+
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-lambda-go/lambda"
+)
+
+func apiGatewayHandler(ctx context.Context, request events.APIGatewayProxyRequest) (*events.APIGatewayProxyResponse, error) {
+	username, err := apigateway.GetUsername(request)
+	if err != nil {
+		return apigateway.LogAndReturnError(err), nil
+	}
+
+	cards, err := storage.LoadUserCards(ctx, random.GetSHA256WithPrefix(clientModel.ClientIDPrefix, username))
```

Comment:
Add the 404 for not found

Reply:
This is not looking to retrieve a specific card, so I think an empty list when there are no cards saved is a more appropriate response as I does right now, what do you think?